pytest_operator/plugin.py

```diff
@@ -98,6 +105,15 @@ def pytest_configure(config):
     )
 
 
+def pytest_runtest_setup(item):
+    if (
+        "skip_if_deployed" in item.keywords
+        and item.config.getoption("--no-deploy")
+        and item.config.getoption("--model") is not None
+    ):
```

Comment:
This is really good idea. But I only found this "ugly" solution.
```python
def pytest_addoption(parser):
    ...
    arg_parser = parser._getparser()
    args = arg_parser.parse_args()
    if args.no_deploy and args.model is None:
        arg_parser.error("model is required")
```

Reply:
doesn't this work:


```suggestion
        item.config.getoption("--no-deploy")
        and item.config.getoption("--model") is None
    ):
        pytest.exit("--model `` is required when --no-deploy is specified")

    if (
        "skip_if_deployed" in item.keywords
        and item.config.getoption("--no-deploy")
        and item.config.getoption("--model") is not None
    ):
```

Reply:
That works, but the output is not really nice.
```bash
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! _pytest.outcomes.Exit: --model is required when --no-deploy is specified !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================================================================================= 2 warnings in 0.19s ==================================================================================================================================================
ERROR: InvocationError for command /home/rg/code/canonical/test-pytest-operator/.tox/integration/bin/pytest --tb native --show-capture=no --log-cli-level=INFO -s --no-deploy tests/integration (exited with code 2)
_______________________________________________________________________________________________________________________________________________________ summary ________________________________________________________________________________________________________________________________________________________
ERROR:   integration: commands failed
```

Reply:
¯\_(ツ)_/¯

i see whatcha mean

Reply:
@addyess @ca-scribner Do you think it's okay to go with that "ugly" solution?

Reply:
Sorry for being so slow in response.  The ugly solution was the best I had in my head, too.  I don't think it is too bad, and it can at least be very clear in the message to the user.  Only nit I'd add is to add to the error message something about --no-deploy, like "must specify --model when using --no-deploy" or similar

Reply:
Ok, I implemented it and here is an [example](https://pastebin.ubuntu.com/p/pKmX86XPJV/) of how it works.
I struggled with unittests a bit, but in the end I found (in the pytest source code) `runpytest_subprocess` as alternative to `runpytest`.

Reply:
Your suggestion is about the error message `must specify --model when using --no-deploy` or do you suggesting to use `pytest.exit` in `pytest_runtest_setup`?

Reply:
oh @r-f-g I can get behind your current solution.  Okay to mark this thread as "resolved"?

Reply:
For me, yes. I'm not sure if @ca-scribner has other suggestions.

Reply:
lgtm!